src/psbt.c

```diff
@@ -1002,3 +1002,521 @@ static int count_psbt_parts(
+            } else if (key_len != 1) {
+                return WALLY_EINVAL;     // Type is more than one byte
+            }
+            p += varint_from_bytes(p, &value_len);
+            clone_bytes(&result->witness_script, p, value_len);
+            result->witness_script_len = value_len;
+
+            p += value_len;
+            break;
+        }
+        case WALLY_PSBT_OUT_BIP32_DERIVATION: {
+            struct keypath_map *keypaths = result->keypaths;
+            size_t path_len;
+            if (key_len != 66 && key_len != 34) {
+                return WALLY_EINVAL;     // Size of key is unexpected
+            }
+            // Check for duplicates
+            for (i = 0; i < keypaths->num_items; ++i) {
+                if (memcmp(keypaths->items[i].pubkey, &key[1], key_len - 1) == 0) {
+                    return WALLY_EINVAL;     // Duplicate key
+                }
+            }
+
+            memcpy(keypaths->items[keypaths->num_items].pubkey, &key[1], key_len - 1);
+
+            // Read the path length
+            p += varint_from_bytes(p, &value_len);
+            if (value_len % 4 != 0 || value_len == 0) {
+                return WALLY_EINVAL;     // Invalid length for keypaths
+            }
+            path_len = (value_len / 4) - 1;
+
+            // Read the fingerprint
+            memcpy(keypaths->items[keypaths->num_items].origin.fingerprint, p, 4);
+            p += 4;
+
+            // Read the path itself
+            keypaths->items[keypaths->num_items].origin.path = wally_malloc(path_len * sizeof(uint32_t));
+            for (i = 0; i < path_len; ++i) {
+                p += uint32_from_le_bytes(p, &keypaths->items[keypaths->num_items].origin.path[i]);
+            }
+            keypaths->items[keypaths->num_items].origin.path_len = path_len;
+
+            keypaths->num_items++;
+            break;
+        }
+        // Unknowns
+        default: {
+            struct unknowns_map *unknowns = result->unknowns;
+            clone_bytes(&unknowns->items[unknowns->num_items].key, key, key_len);
+            unknowns->items[unknowns->num_items].key_len = key_len;
+
+            p += varint_from_bytes(p, &value_len);
+            value = p;
+            clone_bytes(&unknowns->items[unknowns->num_items].value, value, value_len);
+            unknowns->items[unknowns->num_items].value_len = value_len;
+
+            unknowns->num_items++;
+            p += value_len;
+            break;
+        }
+        }
+    }
+
+    if (!found_sep) {
+        return WALLY_EINVAL;
+    }
+
+    *bytes_read = p - bytes;
+    return WALLY_OK;
+}
+
+int wally_psbt_from_bytes(
+    const unsigned char *bytes,
+    size_t bytes_len,
+    struct wally_psbt **output)
+{
+    const unsigned char *p = bytes, *end = bytes + bytes_len, *key, *value;
+    uint64_t key_len, value_len;
+    uint8_t type;
+    size_t i;
+    int ret;
+    struct psbt_counts *counts = NULL;
+    struct wally_psbt *result = NULL;
```

Comment:
What would be the benefit of doing that?

Reply:
otherwise the compiler can't catch when variables are accidently unassigned during refactorings. just a pattern I noticed when working on c-lightning.

Reply:
Not initializing these seems to be causing a segfault. I think it's because one of them isn't being initialized due to another change I am making, and since the pointer isn't null, something is being freed when it shouldn't be.